### PR TITLE
Add com.blunix.currency_converter

### PR DESCRIPTION
**Pull request by tux2linux**
### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by hfiguiere** [comment]
Missing:

>  Please describe your application briefly.

File: com.blunix.git.currency_converter.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  # currency_converter
+  - name: currency_converter
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
+        url: https://git.blunix.com/miscellaneous/currency-converter/uploads/2583116834138e76d8c22e684303cb04/currency_converter-Linux-Portable.tar.gz
```
Comment (hfiguiere) [inline com.blunix.git.currency_converter.yml:27]: how does that work on aarch64?

File: com.blunix.git.currency_converter.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/main/data/flatpak-manifest.schema.json
+
+---
+app-id: com.blunix.git.currency_converter
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
```
Comment (hfiguiere) [inline com.blunix.git.currency_converter.yml:6]: current in 23.08

File: com.blunix.git.currency_converter.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
+        url: https://git.blunix.com/miscellaneous/currency-converter/uploads/2583116834138e76d8c22e684303cb04/currency_converter-Linux-Portable.tar.gz
+        sha256: 5a03e777c3aa7dc61f6fbe9ae44ed50dde057c8a2dd924fbd185b19703fa4b0a
+      - type: dir
```
Comment (hfiguiere) [inline com.blunix.git.currency_converter.yml:29]: don't use `dir` sources.

File: counter_app/assets/icons/com.blunix.git.currency_converter.svg
Comment (hfiguiere) [inline counter_app/assets/icons/com.blunix.git.currency_converter.svg:1]: icon should be managed upstream

File: counter_app/assets/screenshots/screenshot.png
Comment (hfiguiere) [inline counter_app/assets/screenshots/screenshot.png:1]: this doesn't belong here.

File: counter_app/packaging/linux/com.blunix.git.currency_converter.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  <launchable type="desktop-id">com.blunix.git.currency_converter.desktop</launchable>
+  <developer_name>Blunix GmbH</developer_name>
+  <releases>
+    <release version="1.0.1" date="2024-03-29"/>
+  </releases>
+  <url type="homepage">https://git.blunix.com/miscellaneous/currency-converter</url>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://git.blunix.com/miscellaneous/currency-converter/-/raw/main/assets/screenshots/screenshot.png</image>
```
Comment (hfiguiere) [inline counter_app/packaging/linux/com.blunix.git.currency_converter.metainfo.xml:25]: this should not reference the main branch but a specific commit to be immutable.

File: counter_app/packaging/linux/com.blunix.git.currency_converter.metainfo.xml
Comment (hfiguiere) [inline counter_app/packaging/linux/com.blunix.git.currency_converter.metainfo.xml:1]: this file should be managed upstream

File: counter_app/packaging/linux/com.blunix.git.currency_converter.desktop
Comment (hfiguiere) [inline counter_app/packaging/linux/com.blunix.git.currency_converter.desktop:1]: this file should be managed upstream

File: build-flatpak.sh
Comment (hfiguiere) [inline build-flatpak.sh:1]: this is way over complicated and could just be a few build commands in the manifest.

File: counter_app/packaging/linux/com.blunix.git.currency_converter.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      <caption>A screenshot of the app's homepage</caption>
+    </screenshot>
+  </screenshots>
+  <content_rating type="oars-1.0" />
```
Comment (bbhtt) [inline counter_app/packaging/linux/com.blunix.git.currency_converter.metainfo.xml:29]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: com.blunix.git.currency_converter.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/main/data/flatpak-manifest.schema.json
+
+---
+app-id: com.blunix.git.currency_converter
```
Comment (bbhtt) [inline com.blunix.git.currency_converter.yml:4]: Do you own blunix.com or is it just hosting the gitlab instance like gitlab.com?

File: com.blunix.git.currency_converter.yml (hunk 2 above)
Comment (tux2linux) [reply com.blunix.git.currency_converter.yml]: Updated to 23.08

File: com.blunix.git.currency_converter.yml (hunk 6 above)
Comment (tux2linux) [reply com.blunix.git.currency_converter.yml:4]: The app was developed by blunix company that owns blunix.com

File: com.blunix.git.currency_converter.yml (hunk 1 above)
Comment (hfiguiere) [reply com.blunix.git.currency_converter.yml:40]: so what happend in the aarch64 flatpak that is built?

File: com.blunix.git.currency_converter.yml (hunk 6 above)
Comment (hfiguiere) [reply com.blunix.git.currency_converter.yml:4]: then it should be `com.blunix` without the `git. But you say **was**. So it's in the past, who control the domain?

Comment (tux2linux) [reply com.blunix.git.currency_converter.yml:4]: The app was developed by blunix company that owns blunix.com

Comment (tux2linux) [reply com.blunix.git.currency_converter.yml:4]: Blunix does currently control the domain. I decided to add git because the homepage of the app is hosted on git.blunix.com.  
If it aligns better with the flatpak conventions then I will change it to com.blunix. Would you recommend to proceed with the change?

File: com.blunix.git.currency_converter.yml (hunk 1 above)
Comment (tux2linux) [reply com.blunix.git.currency_converter.yml:40]: I have not conducted builds or tests on aarch64 devices as such I cannot provide insight about the apps performance on that architecture. Currently it only supports x86_64 architectures

Comment (hfiguiere) [reply com.blunix.git.currency_converter.yml:40]: that still doesn't answer the question. Let me rephrase again:

There is an aarch64 flatpak being built too. This will not work. You must disable the aarch64 build. This is not done.

File: com.blunix.git.currency_converter.yml (hunk 6 above)
Comment (hfiguiere) [reply com.blunix.git.currency_converter.yml:4]: yes it should be changed.

File: com.blunix.git.currency_converter.yml (hunk 1 above)
Comment (tux2linux) [reply com.blunix.git.currency_converter.yml]: I understood now, apologies for the oversight. 
I added {"only-arches": ["x86_64"] } in flathub.json to disable aarch64 build on flathub.

File: com.blunix.git.currency_converter.yml (hunk 6 above)
Comment (tux2linux) [reply com.blunix.git.currency_converter.yml]: I changed the name. Is it alright or should I create a new pull request from a different branch?

Comment (hfiguiere) [reply com.blunix.git.currency_converter.yml]: no

the branch name is irrelevant.

**Comment by bbhtt** [comment]
bot, build com.blunix.currency_converter
